front-end/src/components/Signup/index.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import styled from 'styled-components';
+import FormInput from '../../shared/FormInput';
+import Logo from '../../svg/Logo';
+
+class Signup extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: '',
+      email: '',
+      password: '',
+      phoneNumber: '',
+      name: '',
+    };
+  }
+
+  handleChange = (field, event) => {
+    this.setState({
+      [field]: event.target.value
+    })
+    console.log(event.target.value);
```

Comment:
Will we be removing any console.logs later down the line?

Reply:
I've only learned a little bit about React but props vs. state was something that tripped me up. Would be cool to get a quick rundown from someone with more experience, maybe during the meeting?

Reply:
Removed the console logs!

Regarding Props vs. State. If you recall back in OOP languages, a class has member variables that can be accessed like so:

```
class MyThing {
    this.prop1 = "hello";
}

React takes a pretty similar approach, where any property in `this.state` is essentially a property of the class that could change as the user interacts with the app. `props` are properties that are passed into the component, sort of like the dependencies you'd inject into a class in OOP. They can be primitive types like numbers or strings, functions, or even other components and are made accessible by calling `super(props)`. 

Here's a full React example:

main.js
```
<Signup firstWord="hello" bar="world"/>
```
in Signup.js
```
class Signup extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      name: "",
      email: "",
    }
  }

  render() {
    return (
       <div>{this.props.firstWord}{this.state.name}</div>
    )
  }
}
```

Reply:
Nice, that's really helpful. Thank you!